metrics-operator/api/v1alpha3/keptnmetric_webhook.go

```diff
@@ -17,11 +17,78 @@ limitations under the License.
 package v1alpha3
 
 import (
+	"time"
+
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
 	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
+// log is for logging in this package.
+var keptnmetriclog = logf.Log.WithName("keptnmetric-resource")
+
 func (r *KeptnMetric) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
 		Complete()
 }
+
+//+kubebuilder:webhook:path=/validate-metrics-keptn-sh-v1alpha3-keptnmetric,mutating=false,failurePolicy=fail,sideEffects=None,groups=metrics.keptn.sh,resources=keptnmetrics,verbs=create;update,versions=v1alpha3,name=vkeptnmetric.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &KeptnMetric{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *KeptnMetric) ValidateCreate() error {
+	keptnmetriclog.Info("validate create", "name", r.Name)
+
+	return r.validateKeptnMetric()
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *KeptnMetric) ValidateUpdate(old runtime.Object) error {
+	keptnmetriclog.Info("validate update", "name", r.Name)
+
+	return r.validateKeptnMetric()
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *KeptnMetric) ValidateDelete() error {
+	keptnmetriclog.Info("validate delete", "name", r.Name)
+
+	return nil
+}
+
+func (s *KeptnMetric) validateKeptnMetric() error {
+	var allErrs field.ErrorList //defined as a list to allow returning multiple validation errors
+	var err *field.Error
+	if err = s.validateRangeInterval(); err != nil {
+		allErrs = append(allErrs, err)
+	}
+	if len(allErrs) == 0 {
+		return nil
+	}
+
+	return apierrors.NewInvalid(
+		schema.GroupKind{Group: "metrics.keptn.sh", Kind: "KeptnMetric"},
+		s.Name,
+		allErrs)
+}
+
+func (s *KeptnMetric) validateRangeInterval() *field.Error {
+	if s.Spec.Range != nil {
+		_, err := time.ParseDuration(s.Spec.Range.Interval)
+		if err != nil {
+			return field.Invalid(
+				field.NewPath("spec").Child("range").Child("interval"),
+				s.Spec.Range.Interval,
+				errors.New("Forbidden! The time interval cannot be parsed. Please check for suitable conventions").Error(),
+			)
+		}
+	}
+	return nil
```

Comment:
to reduce nesting, I suggest to arrange this a little bit
```suggestion
        if s.Spec.Range == nil {
            return nil
        }
	_, err := time.ParseDuration(s.Spec.Range.Interval)
        if err != nil {
	        return field.Invalid(
		        field.NewPath("spec").Child("range").Child("interval"),
		        s.Spec.Range.Interval,
		        errors.New("Forbidden! The time interval cannot be parsed. Please check for suitable conventions").Error(),
	        )
        }
        return nil
```